File: react-native-totvs-camera/android/src/main/java/com/totvs/camera/core/OutputFileOptions.kt
package com.totvs.camera.core

import java.io.File

/**
 * Output options for saving the captured images
 */
data class OutputFileOptions(
    /** output file for a taken picture */
    val file: File,
    /**
     * Indicate whether the saved taken image is reversed horizontally.
     *
     * This can happens when we're using a front camera. By default front cameras
     * have an horizontal mirroring regarding the back camera. We need to set this
     * in order to let the saving engine knows that the image needs to be treated accordingly
     */
    val isReversedHorizontal: Boolean,
    /**
     * * Indicate whether the saved taken image is reversed vertically.
     */
    val isReversedVertical: Boolean
)

File: ios/TOTVSCameraKit/Core/OutputFileOptions.swift
//
//  OutputFileOptions.swift
//  totvs-camera-core
//
//  Created by Jansel Rodriguez on 6/28/20.
//  Copyright © 2020 Jansel. All rights reserved.
//
import Foundation
/**
 * Output options for saving the captured images
 */
public struct OutputFileOptions {
    public static let NULL = OutputFileOptions()
    /** output file for a taken picture */
    public let outputDirectory: URL?
    
    public init(outputDirectory: URL? = nil) {
        self.outputDirectory = outputDirectory
    }
}
